cli/src/lib.rs

```diff
@@ -50,25 +51,95 @@ pub struct Arguments {
     /// Set this flag on the peer that should submit genesis on the network initial start.
     pub submit_genesis: bool,
     /// Set custom genesis file path. `None` if `submit_genesis` set to `false`.
-    pub genesis_path: Option<PathBuf>,
+    pub genesis_path: Option<ConfigPath>,
     /// Set custom config file path.
-    pub config_path: PathBuf,
+    pub config_path: ConfigPath,
 }
 
-const CONFIGURATION_PATH: &str = "config.json";
-const GENESIS_PATH: &str = "genesis.json";
+const CONFIGURATION_PATH: &str = "config";
+const GENESIS_PATH: &str = "genesis";
+const PRECONFIGURED_CONFIG_EXTENSIONS: [&str; 2] = ["json", "json5"];
 const SUBMIT_GENESIS: bool = false;
 
 impl Default for Arguments {
     fn default() -> Self {
         Self {
             submit_genesis: SUBMIT_GENESIS,
-            genesis_path: Some(GENESIS_PATH.into()),
-            config_path: CONFIGURATION_PATH.into(),
+            genesis_path: Some(ConfigPath::new(GENESIS_PATH).with_preconfigured_extensions()),
+            config_path: ConfigPath::new(CONFIGURATION_PATH).with_preconfigured_extensions(),
         }
     }
 }
 
+/// Wrapper around path to config file (i.e. config.json, genesis.json).
+///
+/// Provides abstraction above file extension.
+#[derive(Debug, Clone)]
+pub struct ConfigPath {
+    path: PathBuf,
+    with_preconfigured_extensions: bool,
```

Comment:
Is there any reason we would ever want it set to false?

Reply:
Also supporting this results in slightly confusing API that necessitates the note later

> ```rust
>    /// # Note
>    ///
>    /// After using this function simple `path` provided in [`ConfigPath::new()`] will not be
>    /// counted as a valid path.
>    /// Only `path` with `possible_extensions` will be used in other functions like
>    /// [`ConfigPath::exists()`].
> ```

Reply:
It also seems to me that for now it's more appropriate to have the extension list exhaustive, i.e. failing immediately when extension is neither of the current two. That would make for a more transparent error reporting as well, compared to if it happened further down the line.

Reply:
> Is there any reason we would ever want it set to false?

Yeas, when user provides their own path to config, then extension is clear. This abstraction is needed to find `config.json` or `config.json5` in *default* path

Reply:
> It also seems to me that for now it's more appropriate to have the extension list exhaustive, i.e. failing immediately when extension is neither of the current two

Okay, I'll do it

Reply:
> Also supporting this results in slightly confusing API that necessitates the note later

I was trying to provide an abstraction above path to a configuration. It's either some default expected path with two (or more in future) acceptable extensions or a path provided by user.

So we either add default extensions or not depending on user behaviour.

Don't you like this idea?

Reply:
I'll change API a little bit to make it less confusing

Reply:
> Yes, when user provides their own path to config, then extension is clear. This abstraction is needed to find config.json or config.json5 in default path

Right, sorry, didn't catch that.